service/resource/endpoint/create.go

```diff
@@ -20,8 +21,8 @@ func (r *Resource) GetCreateState(ctx context.Context, obj, currentState, desire
 	}
 
 	createState := Endpoint{
-		ServiceName:      currentEndpoint.ServiceName,
-		ServiceNamespace: currentEndpoint.ServiceNamespace,
+		ServiceName:      desiredEndpoint.ServiceName,
```

Comment:
Small fix as `currentEndpoint` did not have a `serviceName` if the Endpoint didn't exist yet.